crates/relayer/src/chain/axon.rs

```diff
@@ -308,22 +325,25 @@ impl ChainEndpoint for AxonChain {
         self.light_client.check_misbehaviour(update, client_state)
     }
 
-    // FIXME implement this after use a real ics token contract
-    fn query_balance(
-        &self,
-        _key_name: Option<&str>,
-        _denom: Option<&str>,
-    ) -> Result<Balance, Error> {
-        // const DEFAULT_DENOM: &str = "AT";
-
-        // let key_name = key_name.unwrap_or(&self.config.key_name);
-        // let denom = denom.unwrap_or(DEFAULT_DENOM);
-        // let contract = self.ics20_contract()?;
-        // let wallet =  self.get_wallet(key_name);
+    fn query_balance(&self, key_name: Option<&str>, denom: Option<&str>) -> Result<Balance, Error> {
```

Comment:
This only works for source denoms, for sink denoms we would need to lookup the denomTokenContract mapping. (Not saying that it should be implemented now. Just a reminder.)